src/main/java/appeng/container/slot/AppEngCraftingSlot.java

```diff
@@ -195,4 +196,10 @@ public ItemStack decrStackSize( final int par1 )
 
 		return super.decrStackSize( par1 );
 	}
+
+	// TODO: This is really hacky and NEEDS to be solved with a full container/gui refactoring.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)